sycl/include/sycl/detail/defines_elementary.hpp

```diff
@@ -37,9 +37,7 @@
 #endif
 
 #ifndef __SYCL_DEPRECATED
-// The deprecated attribute is not supported in some situations(e.g. namespace)
```

Comment:
I'm not touching lines 60 and 76 because those should be addressed by eliminating the defines and updating the code base to use the C++17 features without wrapping them behind the macros.